datalake_api/querier.py

```diff
@@ -196,7 +210,8 @@ def query_by_time(self, start, end, what, where=None, cursor=None):
             buckets = buckets[i:]
 
         for b in buckets:
-            kwargs = self._prepare_time_bucket_kwargs(b, what)
+            kwargs = self._prepare_time_bucket_kwargs(b, what,
+                                                      limit=MAX_RESULTS/2)
```

Comment:
Ah ok. I guess that's fine, but does it cover the case where you query 2 buckets, have 90 results in the first bucket, and 10 in the second? In my brief readthrough I would expect then it would return 60 results (50 from the first bucket, 10 from the second), potentially masking the remaining 40 in the first bucket.

I'm sure you've already thought about this, but I would expect the behavior to query each bucket with limit=MAX_RESULTS and then truncate your list before giving the response. That's not great either, and I'm probably missing the constraint that got you here (paging perhaps?).